daemon/volumes.go

```diff
@@ -320,6 +321,20 @@ func validMountMode(mode string) bool {
 	return validModes[mode]
 }
 
+func (container *Container) specialMounts() []execdriver.Mount {
+	var mounts []execdriver.Mount
+	if container.ResolvConfPath != "" {
+		mounts = append(mounts, execdriver.Mount{Source: container.ResolvConfPath, Destination: "/etc/resolv.conf", Writable: true, Private: true})
+	}
+	if container.HostnamePath != "" {
+		mounts = append(mounts, execdriver.Mount{Source: container.HostnamePath, Destination: "/etc/hostname", Writable: true, Private: true})
+	}
+	if container.HostsPath != "" {
+		mounts = append(mounts, execdriver.Mount{Source: container.HostsPath, Destination: "/etc/hosts", Writable: true, Private: true})
+	}
```

Comment:
Wondering if there should be a central location to define "special" mounts. "special" handling is also done in https://github.com/docker/docker/blob/b9be50b578f86e858113b9c334e1748e15b63263/daemon/container.go#L991

Perhaps some struct in `container.go` that defines which paths are "special"? This could possibly contain the destination (`/etc/hosts` etc.) as well?

(Just thinking out loud here)